connect/runtime/src/main/java/org/apache/kafka/connect/storage/MemoryOffsetBackingStore.java

```diff
@@ -52,11 +52,15 @@ public void configure(WorkerConfig config) {
 
     @Override
     public synchronized void start() {
+        executor = Executors.newSingleThreadExecutor();
     }
 
     @Override
     public synchronized void stop() {
-        // Nothing to do since this doesn't maintain any outstanding connections/data
+        if (executor != null) {
+            executor.shutdown();
+            executor = null;
```

Comment:
OK will drop synchronization and do awaitTermination(timeout).  @ewencp What do you think is a reasonable timeout before giving up?  Note, get() and set() tasks should be quick but caller's callback could block.